components/src/core/hero/hero.component.ts

```diff
@@ -1,41 +1,119 @@
-import { Component, Input, OnChanges } from '@angular/core';
-export type IconSize = 'normal' | 'large';
+import {
+    AfterContentChecked,
+    AfterViewInit,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Input,
+    OnChanges,
+    ViewChild,
+} from '@angular/core';
+export type IconSize = 'small' | 'normal' | 'large' | number;
 export type FontSize = 'small' | 'normal';
 
 @Component({
     selector: 'pxb-hero',
     template: `
-        <div class="root">
+        <div class="pxb-root">
             <div
-                class="primary-container"
+                class="pxb-primary-wrapper"
+                #primaryContainer
                 [style.backgroundColor]="iconBackgroundColor"
-                [class.large]="iconSize === 'large'"
+                [style.lineHeight.px]="iSize"
+                [style.fontSize.px]="iSize"
+                [style.width.px]="iSize"
+                [style.height.px]="iSize"
+                [class.pxb-svgIcon]="hasMatSvgIcon"
             >
                 <ng-content select="[primary]"></ng-content>
             </div>
-            <span class="channel-value" [style.fontSize.rem]="fontSize === 'small' ? '1' : '1.25'">
-                <ng-content *ngIf="value === undefined" select="pxb-channel-value"></ng-content>
+            <span class="pxb-channel-value-wrapper" [style.fontSize.rem]="fontSize === 'small' ? '1' : '1.25'">
+                <ng-content select="pxb-channel-value"></ng-content>
```

Comment:
does removing this ngif check mean you can pass config through the value attribute and as children? We only want to allow one or the other at a time.

Reply:
Yeah this makes sense, I added the conditional back.